person.rb

```diff
@@ -0,0 +1,45 @@
+require_relative 'nameable'
+require_relative 'rental'
+
+class Person < Nameable
+  attr_reader :id, :parent_permission, :rentals
+  attr_accessor :name, :age
+
+  def initialize(age, parent_permission, name = 'Unknown')
+    super()
+    @id = generate_id
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+    @rentals = []
+  end
+
+  def add_rental(book, date)
+    Rental.new(date, book, self)
+  end
+
+  private
+
+  def generate_id
+    used_ids = Set.new
+    loop do
+      new_id = rand(1000..9999)
+      unless used_ids.include?(new_id)
+        used_ids.add(new_id)
+        return new_id
+      end
+    end
+  end
```

Comment:
- [ ] When I tried the project locally where I tried to create a student, I got the exact same error that the last reviewer @mukhammadrizooff  got which is linked below:

**Error message linked below:**

![image](https://github.com/Armynerh/oop_library/assets/86318284/d2dd6f93-fa9c-44bd-8414-03318cd49a10)

However, on close inspection, I noticed that the fix is actually very simple and can be fixed by adding the "require 'set'" at the top of the file. The error appears because the Set class is not recognized in the project. To use a set in Ruby, you need to require the 'set' library.  ```require 'set'```

Adding this at the top fixed the issue and the project worked as it is supposed to.

- [ ] Kindly fix the error that was mentioned by the previous reviewer by adding the line as I have suggested. Once you are done, please feel free to request for another review in your dashboard.:rocket: